darwin branch fails to build on yosemite

[  7%] Building C object src/onion/CMakeFiles/onion.dir/url.c.o
[  7%] Building C object src/onion/CMakeFiles/onion.dir/poller_libev.c.o
/Users/msoulier/work/mitool-mgmt-proj/onion/src/onion/poller_libev.c:91:2: error:
      'sem_init' is deprecated [-Werror,-Wdeprecated-declarations]
        sem_init(&ret->sem, 0, 1);
        ^
/usr/include/sys/semaphore.h:55:5: note: 'sem_init' has been explicitly marked
      deprecated here
int sem_init(sem_t *, int, unsigned int) __deprecated;
    ^
1 error generated.
make[2]: *** [src/onion/CMakeFiles/onion.dir/poller_libev.c.o] Error 1
make[1]: *** [src/onion/CMakeFiles/onion.dir/all] Error 2
make: *** [all] Error 2

--- a/src/onion/poller_libev.c
+++ b/src/onion/poller_libev.c
@@ -32,7 +32,7 @@
 
 struct onion_poller_t{
 	struct ev_loop *loop;
-	sem_t sem;
+	sem_t *sem;
 	volatile int stop;
 };
 
@@ -88,12 +88,14 @@ void onion_poller_slot_set_type(onion_poller_slot *el, int type){
 onion_poller *onion_poller_new(int aprox_n){
 	onion_poller *ret=calloc(1,sizeof(onion_poller));
 	ret->loop=ev_default_loop(0);
-	sem_init(&ret->sem, 0, 1);
+	ret->sem=sem_open("/poller", O_CREAT, 0, 1);
 	return ret;
 }
 
 /// Frees the poller. It first stops it.
 void onion_poller_free(onion_poller *p){
+	sem_close(p->sem);
+	sem_unlink("/poller");
 	free(p);
 }
 
@@ -152,9 +154,9 @@ void onion_poller_poll(onion_poller *poller){
 
 	poller->stop=0;
 	while(!poller->stop){
-		sem_wait(&poller->sem);
+		sem_wait(poller->sem);
 		ev_run(poller->loop,EVLOOP_ONESHOT);
-		sem_post(&poller->sem);
+		sem_post(poller->sem);
 	}
 }
 /// Stops the polling. This only marks the flag, and should be cancelled with pthread_cancel.

--- a/src/onion/poller_libevent.c
+++ b/src/onion/poller_libevent.c
@@ -33,7 +33,7 @@
 
 struct onion_poller_t{
 	struct event_base *base;
-	sem_t sem;
+	sem_t *sem;
 	volatile int stop;
 };
 
@@ -90,13 +90,15 @@ onion_poller *onion_poller_new(int aprox_n){
 	
 	onion_poller *ret=calloc(1,sizeof(onion_poller));
 	ret->base=event_base_new();
-	sem_init(&ret->sem, 0, 1);
+	ret->sem=sem_init("/poller", O_CREAT, 0, 1);
 	return ret;
 }
 
 /// Frees the poller. It first stops it.
 void onion_poller_free(onion_poller *p){
 	event_base_free(p->base);
+	sem_close(p->sem);
+	sem_unlink("/poller");
 	free(p);
 }
 
@@ -133,9 +135,9 @@ int onion_poller_remove(onion_poller *poller, int fd){
 void onion_poller_poll(onion_poller *poller){
 	poller->stop=0;
 	while(!poller->stop){
-		sem_wait(&poller->sem);
+		sem_wait(poller->sem);
 		event_base_loop(poller->base,EVLOOP_ONCE);
-		sem_post(&poller->sem);
+		sem_post(poller->sem);
 	}
 }
 /// Stops the polling. This only marks the flag, and should be cancelled with pthread_cancel.

